DAOS-16876 tests: test for invalid DTX entry - b26

Nasf-Fan opened the pull request:
DTX logic should be able to detect and handle invalid DTX entry, such as the one has dangling referenced record.

Add new test case to simulate the case of dirty DTX entry was left on disk after commit.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

Nasf-Fan committed:
DAOS-16876 vos: discard invalid DTX when commit or abort - b26

When commit or abort a DTX, we will check whether it is a valid
entry or not. For invalid case, we will discard it with warning
message and increase related metrics counter.

It may be not perfect solution, but it is efficient to help the
user to cleanup system efficiently.

Signed-off-by: Jeff Olivier <jeffolivier@google.com>
Signed-off-by: Fan Yong <fan.yong@hpe.com>

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-15912/3/execution/node/321/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-15912/3/execution/node/259/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-15912/3/execution/node/281/log

daosbuild1 commented:
Test stage Build on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-15912/3/execution/node/438/log

daosbuild1 commented:
Test stage Build on Leap 15.5 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-15912/3/execution/node/476/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-15912/3/execution/node/343/log

Nasf-Fan committed:
DAOS-16876 tests: test for invalid DTX entry - b26

DTX logic should be able to detect and handle invalid DTX entry, such
as the one has dangling referenced record.

Add new test case to simulate the case of dirty DTX entry was left on
disk after commit.

Signed-off-by: Fan Yong <fan.yong@hpe.com>

daltonbohning reviewed (commented):
ftest LGTM

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-15912/6/execution/node/1509/log
